hudi-sync/hudi-adb-sync/src/main/java/org/apache/hudi/sync/adb/HoodieAdbJdbcClient.java

```diff
@@ -0,0 +1,440 @@
+    return doesTableExist(tableName);
+  }
+
+  @Override
+  public Option<String> getLastCommitTimeSynced(String tableName) {
+    String sql = constructShowCreateTableSql(tableName);
+
+    Function<ResultSet, Option<String>> transform = resultSet -> {
+      try {
+        if (resultSet.next()) {
+          String table = resultSet.getString(2);
+          Map<String, String> attr = new HashMap<>();
+          int index = table.indexOf(TBL_PROPERTIES_STR);
+          if (index != -1) {
+            String sub = table.substring(index + TBL_PROPERTIES_STR.length());
+            sub = sub
+                .replaceAll("\\(", "")
+                .replaceAll("\\)", "")
+                .replaceAll("'", "");
+            String[] str = sub.split(",");
+
+            for (String s : str) {
+              String key = s.split("=")[0].trim();
+              String value = s.split("=")[1].trim();
+              attr.put(key, value);
+            }
+          }
+          return Option.ofNullable(attr.getOrDefault(HOODIE_LAST_COMMIT_TIME_SYNC, null));
+        }
+        return Option.empty();
+      } catch (Exception e) {
+        throw new HoodieException("Fail to execute sql:" + sql, e);
+      }
+    };
+    return executeQuerySQL(sql, transform);
+  }
+
+  @Override
+  public void updateLastCommitTimeSynced(String tableName) {
+    // Set the last commit time from the TBLProperties
+    String lastCommitSynced = activeTimeline.lastInstant().get().getTimestamp();
+    try {
+      String sql = constructUpdateTblPropertiesSql(tableName, lastCommitSynced);
+      executeAdbSql(sql);
+    } catch (Exception e) {
+      throw new HoodieHiveSyncException("Fail to get update last commit time synced:" + lastCommitSynced, e);
+    }
+  }
+
+  @Override
+  public Option<String> getLastReplicatedTime(String tableName) {
+    throw new UnsupportedOperationException("Not support getLastReplicatedTime yet");
+  }
+
+  @Override
+  public void updateLastReplicatedTimeStamp(String tableName, String timeStamp) {
+    throw new UnsupportedOperationException("Not support updateLastReplicatedTimeStamp yet");
+  }
+
+  @Override
+  public void deleteLastReplicatedTimeStamp(String tableName) {
+    throw new UnsupportedOperationException("Not support deleteLastReplicatedTimeStamp yet");
+  }
+
+  @Override
+  public void updatePartitionsToTable(String tableName, List<String> changedPartitions) {
+    if (changedPartitions.isEmpty()) {
+      LOG.info("No partitions to change for table:{}", tableName);
+      return;
+    }
+
+    LOG.info("Changing partitions on table:{}, changedPartitionNum:{}", tableName, changedPartitions.size());
+    List<String> sqlList = constructChangePartitionsSql(tableName, changedPartitions);
+    for (String sql : sqlList) {
+      executeAdbSql(sql);
+    }
+  }
+
+  @Override
+  public void dropPartitions(String tableName, List<String> partitionsToDrop) {
+    throw new UnsupportedOperationException("Not support dropPartitions yet.");
```

Comment:
here would we support drop partitions?

Reply:
We don't support drop partitions now, but we will support this feature in the near future.